### ISSUE TITLE
Better management of your owned ships

### ISSUE THREAD

**Issue by esksp** [endless-sky/endless-sky#10592]
### Problem Description

- **Management**: The current system is very limited, and works only if you only have a single companion ship. The system already collapses when you have a companion ship and carry some mining drones: Tell those to mine a specific asteroid, and your warship starts firing too... Now imagine you have mining drones and fighters. Imagine you're in a fight, you'd like to launch those fighters, but is that really a reason to also launch your mining drones (except to quickly lose them)?
- **Fleet AI**: Then there is the player's ships AI issue: You have different ships for different tasks: Your slow, lightly armed cargo ships should be on "timid" (if not "coward") (to use your NPC personalities), while warships should always defend your weaker ships, instead of rushing to fight their own war somewhere on the other side of the system map...
I hate escort mission, when ponderous cargoes slooowly start turning to target their puny lasers instead of running as fast as they can. You tell them to jump away, but by the time they stop pretending to fight and manage to get ready for the jump they're dead... And in the meanwhile your warships, instead of protecting them, are off fighting their own war.

### Related Issue Links

N/A

### Desired Solution

- Management: Please implement a way to give individual orders: Like _you_ stay close, _you_ don't do anything, and so on. Potentially by selecting the ship in question in the escort ship list (bottom left) and then issuing the command?
Also a way to only launch individual fighters/drones (potentially by clicking on their icon?).
- AI: AFAIK NPC ships already have personalities, ideally the player should be able to also set individual personalities for all his non-flagship ships: Aggressive for warships, timid (or pacifist) for cargoes. 
One could overlay a personality letter over the little escort list icons on the bottom left to show the current setting.
There doesn't seem to be a "protector" personality, but it shouldn't be too difficult to implement: "_If_ ship=friend _and_ ship=weak/damaged _and_ ship=under attack, _then_ go attack his attackers".

### Alternative Approaches

none I can think of

**Comment by TomGoodIdea** [comment]
> Management: Please implement a way to give individual orders: Like _you_ stay close, _you_ don't do anything, and so on. Potentially by selecting the ship in question in the escort ship list (bottom left) and then issuing the command?

Isn't it already possible?

> AI: AFAIK NPC ships already have personalities, ideally the player should be able to also set individual personalities for all his non-flagship ships: Aggressive for warships, timid (or pacifist) for cargoes.

#5720

**Comment by TomGoodIdea** [comment]
> > Isn't it already possible?
> 
> Is it?
> AFAIK you only can give _general_ orders (i.e. to _all_ ships). I want to give orders to each ship _individually_. Potentially to each ship a different order.
> Is that possible? I didn't find any mention, and even less the way to do so.

If you have some escorts selected, your orders will be sent only to these escorts, instead of your whole fleet.

> > #5720
> 
> Not sure. From what I understand, #5720 is about hard-coding preferences, while I am talking more about the possibility to change those personalities on the fly, during a battle (ship damaged - Make it flee).
> The problem is that in the game the situation changes constantly, a hard-coded system might be too rigid (although I admit, way better than nothing).
> 
>  Is there an issue with individual player ship personalities (#5720's author hints at that)?

There's a comment on #5720 from one of the main devs, about escort personalities. And there's #802, but it's closed.

**Comment by esksp** [comment]
> If you have some escorts selected, your orders will be sent only to these escorts, instead of your whole fleet.

Oh, great! More secret knowledge... :-D
Seriously, thanks, that's very good to know. Now if there was also a way to only launch _specific_ fighters/drones (as opposed to all a ship carries)...

> There's a comment on #5720 from one of the main devs, about escort personalities.

Yes, already saw that and gave it a thumbs-up. That been said, it's a suggestion from 3 years ago and apparently in limbo. :-(

**Comment by tibetiroka** [comment]
> Now if there was also a way to only launch specific fighters/drones (as opposed to all a ship carries)...

That works the same way. It you have some fighters selected and order them to deploy, only those ships will be deployed.

To make it easier, you can assign groups of escorts to hotkeys. I believe you have to press Control+number to assign them, and the same number to select them afterwards. You can find more info on it on the [wiki page](https://github.com/endless-sky/endless-sky/wiki/PlayersManual#key-bindings-and-shortcuts).

**Comment by esksp** [comment]
> It you have some fighters selected and order them to deploy

Thanks. How do you select fighters while they are docked (invisible)? By clicking on their little icons on the escorts list (bottom left)?

.
Anyway - Sorry guys but this is all a little, well, not very intuitive, and often a little awkward...
Some things do really feel like workarounds, like cases of "let's see how we can do this with what we've already have".  Nothing against optimization, but this is a _game_, the player should be at ease and instinctively know how to do things, _easily_.

Take this "selected" thing for instance, as far as I have understood it so far: 
When I'm in a battle, my current selection is the enemy I'm targeting. Always, obviously.
This means I have to unselect my target (stop fighting), try to select the escort I need (hopefully by the little escort icons, because in the main screen it's very difficult to select ships), give it my orders, then go find and select my opponent again, apologize for the wait, and continue shooting him. Neither fast nor easy.

Yes, you guys might be used to it by now, but it's still very convoluted, acceptable for a spreadsheet, not for real-time combat. It's probably exacerbated by the fact I'm over 60 and my fingers are not as nimble as they used to be, but no matter what, **the UI should be comfortable and intuitive: Luxury cars do the exact same job as cheap ones (go from A to B), the only difference, the reason you pay more, is _how you feel_ when you are inside.**

The game's UI (which I admit is mostly high class) should be _throughout_ smooth, obvious and comfortable. And the combat GUI doubly so, since it's only fractions of seconds between victory and game over: The player should be able to do everything required with a minimum of hassle, those commands should be so simple and obvious you don't need to read the manual to guess them (I do, but most people don't). You might even have a special "Combat HUD" the player can use to keep on top of the situation.

**TL;DR:** While I think the game is absolutely top, I think there is still lots of room for improvement in some parts of the GUI.

My 2 cents worth, feel free to disregard.

**Comment by TomGoodIdea** [comment]
would something like #10208 help?

**Comment by esksp** [comment]
Probably. This or something else, as long as _something_ allows the player to feel like (s)he is in control.

.
That been said, while an active pause would indeed make combat more manageable and less random, the GUI _does_ need improvements here and there.
- What I'm missing most is more **status information** (see #10591).
- Also the **Outfitter GUI** needs streamlining: Inside the Outfitter you have to be able to display things in shop/installed/cargo/storage, and buy/sell/install/uninstall/store. That's 20 combinations you currently have to handle with 3 checkboxes, two buttons, a mouse click combination and a couple hotkeys! At least try to _only_ have checkboxes and buttons.
- Another UI thing missing is some form of **cargo management**: A place you can see on a glance what is in your cargo holds, how much it weights, and allows you to potentially sell/install/dump/put in storage each item (and even move it from one ship to another, if one day "cargo space" stops being a generic shared value).
I've often been trapped because I took off with things in my cargo I didn't even know were there (collected in space without me noticing), to a point I now always go through the Trading Market before leaving a planet, because currently _that_ is the easiest/quickest way to see if there is something in your cargo (the **Information panel** does give that information too, but to get there it's a hotkey, a click, and sifting through a busy list of things. Too long: **There is "immediate" grade interactions, 1-click grade interactions, and the "take your time" grade rest. Managing your cargo is a 1-click interaction** since you do it all the time).

**Comment by esksp** [comment]
> But you can easily discover all those secrets by reading the source code!

No no no! I hope this is a joke, because it's like saying that players should ask the devs what they _intended_ to do, but _didn't succeed_ in doing!...

The fact the code is available is no excuse for anything, people not involved in ES development shouldn't even have to know it's there, much less try to decode it.
For the vast majority of players, when they play ES they just want to be a spaceship pilot. Period.

> I'd say it needs a rewrite from scratch.

That would obviously be better, but I guess it would annoy many people due to them being used to the current form. (And probably trigger some "But that's how EV did this!" nostalgia...)

> Still, a "usecase controlled" UI would be better. 

Probably. But even without that, removing those "quick band-aid solutions" which make it so clumsy would help a lot: For instance, there is no reason asteroid heavy metals trading should happen in the Outfitter. Never mind the nice pics, selling heavy metals belongs to Trading.
Also, separating the "things in cargo" checkbox into "things currently installed" and "things stored in cargo" would spare players a lot of clicking. Right now the game shows both at the same time (why?), and you have to Ctrl-click on your ship to just get a view of what you have in cargo.

If you rewrite it, you would need to also rewrite the Trading panel (which now is kind of pointless, but which could have potential, like additional and more diversified ware to buy/sell), and most of all, create a real cargo management GUI (see recent threads: #10768, #10599, #10717, and PR  #10743).


> That's why I have [autosell](https://gist.github.com/

The solution for unwanted flotsam cluttering your already small cargo bays would be **a hotkey to enable item scooping when you want it, and disable it the rest of the time**. How difficult can this be?...
Unfortunately in the current implementation that feature is hidden 3 clicks down in "Settings" .  :-(
The problem is that once the game has secretly filled up your cargo with junk, it gets very complicated to collect the things you really want. You need to purge the junk, hope whatever you wanted to capture is still around, and that there aren't a dozen enemies hammering you in the meantime...

**Comment by TomGoodIdea** [comment]
Is there any specific feature request here that doesn't duplicate any other issue? If the only thing left is assigning custom keys to toggle preferences, could you please open a new issue focused on just that (assuming one doesn't exist already; I can't find anything)?

**Comment by esksp** [comment]
> could you please open a new issue focused on just that?

You mean me?

Besides, "request here that doesn't duplicate any other issue" is in the eye of the beholder: As the discussion shows, some requests are covered by partial solutions (or something which might pass as a solution), other issues have been mentioned in the past but since forgotten, and so on. So, what is it you actually need?

Existing issues are too vague/undefined (IMHO).
If it's a clear list of potential issues to address/things to improve you're looking for, I think a new, specific hierarchical list by category might be more efficient. Something like:
```
1. - Player interface:
  1.1 - General game interface (load/save/new game, etc.)
  1.2 - Travel interface (map, useful info, route planning)
  1.3 - Combat interface
    1.3.1 - Fleet combat interface (immediate orders to individual ships, like "flee!" or "defend that ship!")
  1.4 - Fleet management interface (parking, orders and general AI settings for travel/mining/combat) 
  1.5 - Flagship management interface (equipment, status etc.)
  1.6 - Cargo management interface (including selling/discarding)

(and so on)
```
You immediately see the issues I was talking about in my first post are partly in 1.4, partly in 1.3.1, and while I'm not a developer, I assume they would be dealt with in different parts of the code, which means that although they both are about escort management, they may need to be treated as separate issues, isn't it?

**Comment by TomGoodIdea** [comment]
A feature request _issue_ should describe a single (or more if they are closely related) problem that can be solved and recognized as solved. A discussion about various aspects of the game interface doesn't have any objective solution, as you could always find new problems, so a _discussion_ would be more fitting.
Feel free to comment on other open issues (even older ones) if your suggestions duplicate them and you want to contribute to finding solutions. Open a discussion if you want to talk about the interfaces in general, or an issue if you have any change requests not covered by other issues (including the ones you mentioned). If this issue's original problems have been solved, it should be closed.

**Comment by esksp** [comment]
Thanks for clarifying.

**Comment by TomGoodIdea** [comment]
Closing this as completed, then. Individual orders are possible, and ship personalities have their own issue.
